Homework/homework2/task4.py

```diff
@@ -0,0 +1,28 @@
+"""
+Write a function that accepts another function as an argument.
+
+Then it should return such a function, so the every call to initial one
+should be cached.
+def func(a, b):
+    return (a ** b) ** 2
+cache_func = cache(func)
+some = 100, 200
+val_1 = cache_func(*some)
+val_2 = cache_func(*some)
+assert val_1 is val_2
+"""
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    cache_memory = []
+
+    def caching(*args, **kwargs):
+        for stored_args, result in cache_memory:
+            if stored_args == (args, kwargs):
```

Comment:
Try:
```
a = [1,2,3]
foo(a)
a.append(4)
foo([1,2,3])
```
I don't expect a solution, just tell me 2 things:
1) why it will fail
2) which python module can help soving it with one call?

Reply:
If I correctly understood
1. List is mutable and so "a" will be changed in cache memory after appending.
2. I really don't know(

Reply:
> I really don't know(

Ask Google how to make a full copy of the list.

Reply:
Via copy module. We may make a copy of "a" and change it how we want without changing the list in the cache memory.

Reply:
Please point to the exact method in the Python documentation.

Reply:
copy.copy(obj)
or sets, dicts and lists have a method .copy() and there is no need to import copy module

Reply:
By pointing I mean a link to the documentation.  https://docs.python.org/3/library/copy.html#copy.copy

`copy` can solve this particular case, but I want a method that solve all cases.

Reply:
https://docs.python.org/3/library/copy.html#copy.deepcopy
If the obj is smth more complex like list of lists and etc.